pkg/metallb/metallb_test.go

```diff
@@ -366,6 +366,32 @@ func TestMetalLbWithOptions(t *testing.T) {
 	assert.Equal(t, "error", testBuilder.errorMsg)
 }
 
+func TestMetalLbWithFRRconfigAlwaysBlock(t *testing.T) {
+	testCases := []struct {
+		testMetalLb   *Builder
+		prefixes      []string
+		expectedError string
+	}{
+		{
+			testMetalLb:   buildValidMetalLbBuilder(buildMetalLbTestClientWithDummyObject()),
+			expectedError: "",
+			prefixes:      []string{"192.168.1.0/24"},
+		},
+		{
+			testMetalLb:   buildValidMetalLbBuilder(buildMetalLbTestClientWithDummyObject()),
+			expectedError: "The prefix 192.168.1. is not a valid CIDR",
+			prefixes:      []string{"192.168.1."},
+		},
+	}
+
+	for _, testCase := range testCases {
+		assert.Equal(t, defaultMetalLbNodeSelector, testCase.testMetalLb.Definition.Spec.SpeakerNodeSelector)
+		assert.Nil(t, nil, testCase.testMetalLb.Object)
```

Comment:
I don't think this assert is necessary, but also assert.Nil is of the form `func Nil(t TestingT, object interface{}, msgAndArgs ...interface{})` so you'd want to do `assert.Nil(t, testCase.testMetalLb.Object)`

Reply:
removed assert nil